paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerAdapter.kt

```diff
@@ -180,33 +175,27 @@ interface CustomerAdapter {
     }
 
     @ExperimentalCustomerSheetApi
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     sealed class Result<T> {
 
         @ExperimentalCustomerSheetApi
-        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
         class Success<T> internal constructor(
             val value: T
         ) : Result<T>()
 
         @ExperimentalCustomerSheetApi
-        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
         class Failure<T> internal constructor(
             val cause: Throwable,
             val displayMessage: String? = null
         ) : Result<T>()
 
         @ExperimentalCustomerSheetApi
-        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
         companion object {
             @ExperimentalCustomerSheetApi
-            @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
             fun <T> success(value: T): Success<T> {
```

Comment:
We should decide if we want to have the classes public or the methods, but I don’t think we want to expose both.

Reply:
The constructors of the public classes are internal, is that sufficient? Also, if we decide to make the classes internal, then we need to change the shape of the success/failure methods to return the base class, since we would be exposing the internal class.

Reply:
We didn't use kotlin result for a few reasons, first, we want to support java, second, we wanted to expose a displaymessage for failures. We'll need to figure out a new way to expose the display message along with the exception for this.

Reply:
Discussed over zoom call. This would be a large change, we are basically doing a hybrid of kotlin.Result and the existing pattern we have today (e.g. PaymentSheetResult). We will stick with the existing pattern and in the next major version bump, revisit this.